e2e/LongHaul/device/IotHub.cs

```diff
@@ -227,6 +231,47 @@ await _deviceClient
             _logger.Trace($"Set the reported property with name [{keyName}] in device twin.", TraceSeverity.Information);
         }
 
+        public async Task UploadFilesAsync(CancellationToken ct)
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                string fileName = $"TestPayload-{Guid.NewGuid()}.txt";
+                using var ms = new MemoryStream(Encoding.UTF8.GetBytes("TestPayload"));
+                _logger.Trace($"Uploading file {fileName}");
+
+                var fileUploadSasUriRequest = new FileUploadSasUriRequest(fileName);
+                FileUploadSasUriResponse sasUri = await _deviceClient.GetFileUploadSasUriAsync(fileUploadSasUriRequest).ConfigureAwait(false);
+                Uri uploadUri = sasUri.GetBlobUri();
+
+                try
+                {
+                    _logger.Trace($"Attempting to upload {fileName}...");
+                    var blockBlobClient = new BlockBlobClient(uploadUri);
+                    await blockBlobClient.UploadAsync(ms, new BlobUploadOptions()).ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    _logger.Trace($"Exception occurred while using Azure Storage SDK to upload file: {ex}", TraceSeverity.Warning);
+                    var failedFileUploadCompletionNotification = new FileUploadCompletionNotification(sasUri.CorrelationId, false)
+                    {
+                        StatusCode = 500,
+                    };
+
+                    await _deviceClient.CompleteFileUploadAsync(failedFileUploadCompletionNotification);
+                    return;
+                }
+
+                _logger.Trace("File upload to Azure Storage was a success");
+                var successfulFileUploadCompletionNotification = new FileUploadCompletionNotification(sasUri.CorrelationId, true)
+                {
+                    StatusCode = 200,
+                };
+
+                await _deviceClient.CompleteFileUploadAsync(successfulFileUploadCompletionNotification, ct);
+                await Task.Delay(5 * 1000);
```

Comment:
```suggestion
                await Task.Delay(5 * 1000, ct).ConfigureAwait(false);
```

Reply:
This probably needs to be in a finally block. I'd also suggest adding a catch with generic Exception and log that as an error.

Reply:
also, I'd suggest to use a variable to specify the file uploading operations interval, similar to what we did for `s_messageLoopSleepTime`.